### ISSUE TITLE
Ha-Proxy Unable to be requested normally

### ISSUE THREAD

**Issue by shouziws** [haproxy/haproxy#2297]
### Detailed Description of the Problem

HA-Proxy version 1.5.18 2016/05/10
I use nginx proxy to HA-Proxy.  What is the reason why the port occupied by haproxy cannot be accessed after a period of time

![image](https://github.com/haproxy/haproxy/assets/33045888/e3658cac-86ca-4144-9f96-1cc98d215356)

### Expected Behavior

The HA-Proxy's port is being accessed normally

### Steps to Reproduce the Behavior

HA-Proxy version 1.5.18 2016/05/10
I use nginx proxy to HA-Proxy.  What is the reason why the port occupied by haproxy cannot be accessed after a period of time

![image](https://github.com/haproxy/haproxy/assets/33045888/e3658cac-86ca-4144-9f96-1cc98d215356)

### What is your configuration?

```haproxy
This is my HA-Proxy config content
![image](https://github.com/haproxy/haproxy/assets/33045888/784b5025-fe3d-475c-a49e-cdd7e64d3352)

This is my Nginx config
![image](https://github.com/haproxy/haproxy/assets/33045888/57bef742-8ff3-43a4-ba98-f751892419d5)
![image](https://github.com/haproxy/haproxy/assets/33045888/d6716c7f-aa85-4adb-984d-96feb4f4d746)
```

### Output of `haproxy -vv`

```plain
[root@k8s-master-3 conf]# haproxy -vv
HA-Proxy version 1.5.18 2016/05/10
Copyright 2000-2016 Willy Tarreau <willy@haproxy.org>

Build options :
  TARGET  = linux2628
  CPU     = generic
  CC      = gcc
  CFLAGS  = -O2 -g -fno-strict-aliasing -DTCP_USER_TIMEOUT=18
  OPTIONS = USE_LINUX_TPROXY=1 USE_GETADDRINFO=1 USE_ZLIB=1 USE_REGPARM=1 USE_OPENSSL=1 USE_PCRE=1

Default settings :
  maxconn = 2000, bufsize = 16384, maxrewrite = 8192, maxpollevents = 200

Encrypted password support via crypt(3): yes
Built with zlib version : 1.2.7
Compression algorithms supported : identity, deflate, gzip
Built with OpenSSL version : OpenSSL 1.0.2k-fips  26 Jan 2017
Running on OpenSSL version : OpenSSL 1.0.2k-fips  26 Jan 2017
OpenSSL library supports TLS extensions : yes
OpenSSL library supports SNI : yes
OpenSSL library supports prefer-server-ciphers : yes
Built with PCRE version : 8.32 2012-11-30
PCRE library supports JIT : no (USE_PCRE_JIT not set)
Built with transparent proxy support using: IP_TRANSPARENT IPV6_TRANSPARENT IP_FREEBIND

Available polling systems :
      epoll : pref=300,  test result OK
       poll : pref=200,  test result OK
     select : pref=150,  test result OK
Total: 3 (3 usable), will use epoll.


[root@k8s-master-3 conf]# uname -a
Linux k8s-master-3 4.19.12-1.el7.elrepo.x86_64 #1 SMP Fri Dec 21 11:06:36 EST 2018 x86_64 x86_64 x86_64 GNU/Linux
```

**Comment by git001** [comment]
I don't see any indicator for this statment in the visible screenshots as a 404 is a normal response.

>What is the reason why the port occupied by haproxy cannot be accessed after a period of time

## Some questions
What was changed before you recognized that behavior?
What's the relevant output of journalctl?
What's in the logs of haproxy, nginx and kernel?

## tips for gh issue
Please post the configs as text and full configs, keep in mind to remove or mask sensitive data.

## Some comments about the setup.

The HAProxy Version is way outdated even unsupported since "2020-01-10 (unmaintained)" https://www.haproxy.org/ . Consider to update to a recent version like 2.8 .

Looks like you use Kubernetes which brings me to the assumption that nginx or haproxy is used as ingress controller, how about to give https://github.com/haproxytech/kubernetes-ingress a chance?.

Looks like that the kernel is also quite old as I can't find it in the el repo https://elrepo.org/linux/kernel/el7/x86_64/RPMS/

**Comment by wtarreau** [comment]
Honestly, branch 1.5 has reached end of life almost 4 years ago, and in practice the last fix was issued 7 years ago. Your version is not even the last one for this branch, and according to what I'm seeing on the bugs page ( http://www.haproxy.org/bugs/bugs-1.5.18.html ) it might even be affected by some close_wait issues that were fixed in 1.5.19, and that could be the cause of what you're seeing. If you're feeling really brave you may want to give 1.5.19 a try, but don't expect any support on it. As Alex said above, you should really upgrade to a more recent version. There will be much less bugs even on a very recent version than what has been accumulating on top of unmaintained 1.5 (not counting vulnerabilities!). If you're worried about upgrading to a too recent version, 2.4 or 2.6 may be considered good choices (2 and 1 year-old respectively).

**Comment by capflam** [comment]
The issue seems to be inactive, I'm closing.
